### DISCUSSION TITLE
Cypress refuses to update the binary to the proper version

### DISCUSSION THREAD

**Discussion by mcv** [cypress-io/cypress#14507] [Questions and Help]
Cypress is being surreal. `npm install` clearly doesn't update cypress properly, because the binary version is woefully out of date. Various variations of `npm install cypress` (-g, --save-dev, with or without version number) accomplish nothing.

```>cypress -v
Cypress package version: 6.2.1
Cypress binary version: 3.3.1
Electron version: not found
Bundled Node version: not found
```

```>cypress install
Cypress 3.3.1 is installed in C:\Users\me\AppData\Local\Cypress\Cache\6.2.1
Skipping installation:
  Pass the --force option if you'd like to reinstall anyway.
```

```>cypress install --force
Cypress 3.3.1 is installed in C:\Users\me\AppData\Local\Cypress\Cache\6.2.1

‼ Warning: Forcing a binary version different than the default.

  The CLI expected to install version: 6.2.1

  Instead we will install version: c:\tools\cypress.zip

  These versions may not work properly together.

  ×  Unzipping Cypress      
    → Cypress Version: 6.2.1
The Cypress App could not be unzipped.

Search for an existing issue or open a GitHub issue at

https://github.com/cypress-io/cypress/issues
----------
Error: EBUSY: resource busy or locked, unlink 'C:\Users\me\AppData\Local\Cypress\Cache\6.2.1\Cypress\blink_image_resources_200_percent.pak'
----------
Platform: win32 (10.0.18363)
Cypress Version: 6.2.1
```

```>cypress -v
Cypress package version: 6.2.1
Cypress binary version: not installed
Electron version: not found
Bundled Node version: not found
```

So what is the proper way to update the binary version of Cypress?

**Comment by bahmutov** [comment]
Can you install the Cypress v3.3.1 first and then install the new version of Cypress

**Reply by mcv** [reply]
It looks like my update removed version 3.3.1. No idea why.

I have some idea of what happened, though. I'm behind a corporate proxy. I just followed the instructions on https://docs.cypress.io/guides/references/proxy-configuration.html#Windows and they don't work. I think that when I installed 3.3.1 long ago, I worked around this problem by downloading it in a zip and setting a variable so cypress knows where to find it. That worked, except Cypress now finds the old version there, which explains that error message.

Unsetting the variable leads to complaints about the proxy again. I set the correct proxy env variable, but it doesn't work.

**Reply by RavinderNuvei** [reply]
@mcv did you find the solution to this problem ? I am also in the similar situation, I am also behind the corporate proxy and I installed the cypress by setting a variable .

I want to install the cypress 13.11.0 but expected package version is 13.13.0.

⚠ Warning: Binary version 13.11.0 does not match the expected package version 13.13.0

  These versions may not work properly together.


Any response would be appreciated.

**Reply by MikeMcC399** [reply]
@RavinderNuvei 
You can try

```shell
npm install cypress@13.11.0 -D
```

**Reply by RavinderNuvei** [reply]
@MikeMcC399 

Thanks for the input, as I am behind the corporate proxy direct commands to install cypress are not working for me :

<img width="480" alt="image" src="https://github.com/user-attachments/assets/205cb0ce-c613-4b4a-a948-a990d41f70e0">

When I am using npm install cypress@13.11.0 -D , it is giving me certificate issues , as in the image 

Because of this , I tried installing cypress 13.11.0 by downloading zip file and setting up a variable. 

I don't have any problem running the tests, it is just that I get this warning message every time I am running my test cases. I just don't want that warning to appear.

any more suggestions would be really appreciated 

@mcv @bahmutov

**Reply by MikeMcC399** [reply]
@RavinderNuvei 

- You can find a very similar issue on Windows discussed in https://github.com/cypress-io/cypress/issues/29620 with `Error: unable to get local issuer certificate`. Take a look and see if any of the workarounds and suggestions are helpful to you.
